src/hotspot/share/interpreter/oopMapCache.cpp

```diff
@@ -541,8 +542,8 @@ void OopMapCache::lookup(const methodHandle& method,
   }
 
   // First search for an empty slot
-  for(i = 0; i < _probe_depth; i++) {
-    entry = entry_at(probe + i);
+  for (int i = 0; i < _probe_depth; i++) {
```

Comment:
I don't think so: GlobalCounter guards against the reclamation of `OopMapCacheEntry`-es, so we only need to protect the paths that access their contents. We don't need it for anything else, like just poking into the array slots. I used to have the critical section that spans this entire method, but reasoned it was excessive.

Reply:
Ok, I concur.  It looks ok since the oopMapCache is an array so the search for the null container won't search through next pointers of entries that could be removed.